internal/git/gitsync_processor.go

```diff
@@ -601,7 +594,17 @@ func NewGitSyncProcessor(ctx context.Context, gitSync *v1alpha1.GitSync, kubeCli
 	}
 	localRepoPath := getLocalRepoPath(gitSync.GetName())
 	go func(repo *v1alpha1.RepositoryPath) {
-		r, err := cloneRepo(ctx, localRepoPath, repo)
+		globalConfig, err := controllerconfig.GetConfigManagerInstance().GetConfig()
+		if err != nil {
+			logger.Errorw("error getting  the  global config", "err", err)
+		}
+		gitCredentials := gitshared.FindCredByUrl(repo.RepoUrl, globalConfig)
+		cloneOptions, err := gitshared.GetRepoCloneOptions(ctx, gitCredentials, kubeClient, namespace, repo)
```

Comment:
If prefix is not found, `gitCredentials` will be nil and `GetRepoCloneOptions()` doesn't seem to be able to handle that.

Reply:
Ok i will put few more tests for them .